src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveCreateTrueMaskTest.template

```diff
@@ -0,0 +1,268 @@
+            }
+            else
+            {
+                // Validates we throw on unsupported hardware
+                test.RunUnsupportedScenario();
+            }
+
+            if (!test.Succeeded)
+            {
+                throw new Exception("One or more scenarios did not complete as expected.");
+            }
+        }
+    }
+
+    public sealed unsafe class CreateTrueMaskTest__{TestName}
+    {
+        private static int FloorPow2(int x)
+        {
+            if (x <= 0)
+            {
+                throw new Exception("Expected a positive argument value.");
+            }
+
+            int result = 1;
+
+            for (x = x / 2; x > 0; x /= 2)
+            {
+                result <<= 1;
+            }
+
+            return result;
+        }
+
+        // DecodePredCount()
+        // =================
+        // integer DecodePredCount(bits(5) bitpattern, integer esize)
+        //     integer elements = CurrentVL DIV esize;
+        //     integer numElem;
+        //     case bitpattern of
+        //         when '00000' numElem = FloorPow2(elements);
+        //         when '00001' numElem = if elements >= 1 then 1 else 0;
+        //         when '00010' numElem = if elements >= 2 then 2 else 0;
+        //         when '00011' numElem = if elements >= 3 then 3 else 0;
+        //         when '00100' numElem = if elements >= 4 then 4 else 0;
+        //         when '00101' numElem = if elements >= 5 then 5 else 0;
+        //         when '00110' numElem = if elements >= 6 then 6 else 0;
+        //         when '00111' numElem = if elements >= 7 then 7 else 0;
+        //         when '01000' numElem = if elements >= 8 then 8 else 0;
+        //         when '01001' numElem = if elements >= 16 then 16 else 0;
+        //         when '01010' numElem = if elements >= 32 then 32 else 0;
+        //         when '01011' numElem = if elements >= 64 then 64 else 0;
+        //         when '01100' numElem = if elements >= 128 then 128 else 0;
+        //         when '01101' numElem = if elements >= 256 then 256 else 0;
+        //         when '11101' numElem = elements - (elements MOD 4);
+        //         when '11110' numElem = elements - (elements MOD 3);
+        //         when '11111' numElem = elements;
+        //         otherwise    numElem = 0;
+        //     return numElem;
+
+        private static int DecodePredCount({Op1Type} bitpattern, int esize)
+        {
+            if (bitpattern > {Op1Type}.All)
+            {
+                throw new Exception("Bit patterns must be 5 bits long.");
+            }
+
+            switch (esize) {
+                case 8:
+                case 16:
+                case 32:
+                case 64:
+                    break;
+                default:
+                    throw new Exception("Incorrect esize");
+            }
+
+            int elements = (int)(Sve.Count64BitElements()) * 64 / esize;
```

Comment:
nit:
```suggestion
            int elements = (int)(Sve.Count8BitElements() / esize);
```

Reply:
It would have to be `int elements = (int)(Sve.Count8BitElements()) * 8 / esize;` anyway I suppose? Or do you suggest to make `esize` denote the size of elements in bytes as well? At the moment it's in bits, not bytes.

Reply:
yes, should have `* 8` and that gives better readability in terms of "X bytes".

Reply:
Done.